### PR TITLE
test: add a simple signer test

### PR DESCRIPTION
**Pull request by iljakuklic**
# Description

Adding a very basic signer test of questionable value. It only checks the wallet signer for now but should be easy to extend to check GCP KMS and other signers too.

## PR Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added or updated tests that comprehensively prove my change is effective or that my feature works

### CODE REVIEW

File: crates/agglayer-signer/src/tests.rs [hunk 1]
```diff
@@ -0,0 +1,43 @@
+use ethers::types::{Eip1559TransactionRequest, H160};
+
+use super::*;
+
+async fn check_signer_works(signer: &ConfiguredSigner) {
```
Comment (Freyskeyd) [inline crates/agglayer-signer/src/tests.rs:5]: Why put this in a dedicated function and not directly in the test function?

**Review by Freyskeyd — APPROVED** [review]

File: crates/agglayer-signer/src/tests.rs (hunk 1 above)
Comment (iljakuklic) [reply crates/agglayer-signer/src/tests.rs:5]: The idea is to have separate tests for different kinds of signers (wallet, gcp kms, etc) but reuse the test logic.

Comment (Freyskeyd) [reply crates/agglayer-signer/src/tests.rs:5]: Okey. at some point we could rely on rs-test fixture then

Comment (iljakuklic) [reply crates/agglayer-signer/src/tests.rs]: > Okey. at some point we could rely on rs-test fixture then

I ported the test to rstest to prepare for this.

**Review by Freyskeyd — APPROVED** [review]

**Pull request merged**